lib/fog/storage/google_json/requests/put_object.rb

```diff
@@ -46,11 +46,14 @@ def put_object(bucket_name,
                        kms_key_name: nil,
                        predefined_acl: nil,
                        **options)
+          if data.is_a?(String)
+            data = StringIO.new(data)
+          end
+
           unless options[:content_type]
-            if data.is_a?(String)
-              data = StringIO.new(data)
-              options[:content_type] = "text/plain"
-            elsif data.is_a?(::File)
+            options[:content_type] = "text/plain"
```

Comment:
Why we specify `content_type` plain then? I would rather say that default should be `application/octet-stream`.